shared/src/main/kotlin/org/jetbrains/kotlin/konan/target/Linker.kt

```diff
@@ -136,19 +136,22 @@ open class MacOSBasedLinker(targetProperties: AppleConfigurables)
     private val dsymutil = "$absoluteLlvmHome/bin/llvm-dsymutil"
 
     private fun provideCompilerRtLibrary(libraryName: String): String? {
-        val suffix = if (libraryName.isNotEmpty() && target == KonanTarget.IOS_X64) {
-            "iossim"
+        val prefix = when (val family = target.family) {
```

Comment:
Why not just have `when` over targets?
This would be simpler and more discoverable.